processor/src/main/resources/org/mapstruct/ap/internal/model/source/builtin/CalendarToXmlGregorianCalendar.ftl

```diff
@@ -27,7 +27,10 @@ private <@includeModel object=findType("XMLGregorianCalendar")/> ${name}( <@incl
     try {
         <@includeModel object=findType("GregorianCalendar")/> gcal = new <@includeModel object=findType("GregorianCalendar")/>( cal.getTimeZone() );
         gcal.setTimeInMillis( cal.getTimeInMillis() );
-        return <@includeModel object=findType("DatatypeFactory")/>.newInstance().newXMLGregorianCalendar( gcal );
+        if ( ${supportingField.variableName} == null ) {
```

Comment:
Should go into the constructor as dicussed and be made final.

Reply:
yep